tests/common.go

```diff
@@ -5311,3 +5318,83 @@ func IsEksPxOperator() bool {
 	}
 	return false
 }
+
+// ConcurrentExecute executes the given operation on each item in the collection, either sequentially
+// or in parallel, depending on the specified execution mode.
+//
+// Parameters:
+//
+//	items: The collection of items to operate on (slice, array, or map).
+//	operation: The function to execute on each item. If the function takes one argument,
+//	           it will be passed the item value. If it takes two arguments, the first
+//	           will be the item key or index, and the second will be the item value.
+//	executionMode: The mode to use for executing the operation, either "Sequential" or "Parallel".
+func ConcurrentExecute(items interface{}, operation interface{}, executionMode ExecutionMode) {
+	v := reflect.ValueOf(items)
+	var keys []reflect.Value
+	if v.Kind() == reflect.Map {
+		keys = v.MapKeys()
+	} else if v.Kind() == reflect.Slice || v.Kind() == reflect.Array {
+		keys = make([]reflect.Value, v.Len())
+		for i := 0; i < v.Len(); i++ {
+			keys[i] = v.Index(i)
+		}
+	} else {
+		panic(fmt.Sprintf("Instead of %#v, items should be a slice, array, or map.", v.Kind().String()))
```

Comment:
instead of panic can we throw error here

Reply:
Discarded the function because it was not needed in this test instance.